source/libs/cache.ts

```diff
@@ -1,73 +1,33 @@
-type CacheGetter<TValue extends any = any> = () => TValue | Promise<TValue>;
+const storage: browser.storage.StorageArea = browser.storage.local;
 
-interface CacheRequest {
-	code: string;
-	key: string;
-	value?: unknown;
-	expiration?: number;
-}
-
-export async function getSet<TValue extends any = any>(key: string, getter: CacheGetter<TValue>, expiration?: number): Promise<TValue | undefined> {
-	const cache = await get<TValue>(key);
-	if (cache !== undefined) {
-		return cache;
-	}
-
-	const value = await getter();
-	if (value !== undefined) {
-		await set(key, value, expiration);
-	}
-
-	return value;
+export async function has(key: string): Promise<boolean> {
+	const values = await storage.get(key);
+	return values[key] !== undefined;
 }
 
 export async function get<TValue extends any = any>(key: string): Promise<TValue | undefined> {
-	const value = await browser.runtime.sendMessage({
-		key,
-		code: 'get-cache'
-	});
-
+	const values = await storage.get(key);
+	const value = values[key];
 	// If it's not in the cache, it's best to return "undefined"
 	if (value === null || value === undefined) {
 		return undefined;
 	}
 
-	return value;
+	console.log('CACHE: found', key, value.data);
+	return value.data;
 }
 
 export function set<TValue extends any = any>(key: string, value: TValue, expiration?: number /* in days */): Promise<any> {
-	return browser.runtime.sendMessage({
-		key,
-		value,
-		expiration,
-		code: 'set-cache'
+	console.log('CACHE: setting', key, value);
+	return storage.set({
+		[key]: {
+			data: value,
+			lastUse: Date.now(),
```

Comment:
I think we should make the expiration time a required parameter.
And since we always have that we don't need the `lastUse`